sklearn/metrics/classification.py

```diff
@@ -1994,3 +1994,123 @@ def brier_score_loss(y_true, y_prob, sample_weight=None, pos_label=None):
     y_true = np.array(y_true == pos_label, int)
     y_true = _check_binary_probabilistic_predictions(y_true, y_prob)
     return np.average((y_true - y_prob) ** 2, weights=sample_weight)
+
+
+def calibration_loss(y_true, y_prob, reducer="sum", nbins=10, normalize=True,
+                     sample_weight=None, pos_label=None):
+    """Compute  calibration loss.
+
+    Across all items in a set N predictions, the calibration loss measures
+    the aggregated difference between (1) the average predicted probability
+    assigned to the possible outcomes for item i, and (2) the frequencies
+    of the actual outcome.
+    Therefore, the lower the calibration loss is for a set of predictions, the
+    better the predictions are calibrated.
+
+    The aggregation method can be either:
+    - 'sum' for sum_k P_k delta_k (default), denoted as ECE in [1]
+    - 'max' for max_k delta_k, denoted as MCE in [1]
+    where k denotes a bin number and delta_k = abs(avg_freq_true_k - center_k)
+
+    The calibration loss is appropriate for binary and categorical outcomes
+    that can be structured as true or false, but is inappropriate for ordinal
+    variables which can take on three or more values.
+    Which label is considered to be the positive label is controlled via the
+    parameter pos_label, which defaults to 1.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    ----------
+    y_true : array, shape (n_samples,)
+        True targets.
+
+    y_prob : array, shape (n_samples,)
+        Probabilities of the positive class.
+
+    reducer: string, must be among 'l1', 'l2', 'max'
+        Aggregation method.
+
+    nbins: int, positive, optional (default=10)
```

Comment:
Need space before colon. I also think `n_bins` or just `bins` might be more conventional.